pkg/gateway/middleware.go

```diff
@@ -198,29 +199,22 @@ func OperationLookupHandler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		ctx := req.Context()
 		o := ctx.Value(ContextKeyOperation).(*operations.Operation)
-		repoID := ctx.Value(ContextKeyRepositoryID).(string)
 		o.OperationID = operations.OperationIDOperationNotFound
-		if repoID == "" {
-			if req.Method == http.MethodGet {
-				o.OperationID = operations.OperationIDListBuckets
-			} else {
-				_ = o.EncodeError(w, req, nil, gatewayerrors.ERRLakeFSNotSupported.ToAPIErr())
-				return
-			}
-		} else {
-			ref := ctx.Value(ContextKeyRef).(string)
-			pth := ctx.Value(ContextKeyPath).(string)
-			switch {
-			case ref != "" && pth != "":
-				req = req.WithContext(ctx)
-				o.OperationID = pathBasedOperationID(req.Method)
-			case ref == "" && pth == "":
-				o.OperationID = repositoryBasedOperationID(req.Method)
-			default:
-				w.WriteHeader(http.StatusNotFound)
-				return
-			}
+
+		repoID := ctx.Value(ContextKeyRepositoryID).(string)
+		ref := ctx.Value(ContextKeyRef).(string)
+		pth := ctx.Value(ContextKeyPath).(string)
+
+		// based on the operation level, we can determine the operation id
+		switch {
+		case repoID == "":
+			o.OperationID = rootBasedOperationID(req.Method)
+		case ref != "" && pth != "":
+			o.OperationID = pathBasedOperationID(req.Method)
+		case ref == "" && pth == "":
+			o.OperationID = repositoryBasedOperationID(req.Method)
```

Comment:
I think we're missing a `default` case here!  Even if we keep going in order to fail later, I would _really_ like a log (level TRACE) for each of the _four_ cases.  Otherwise debugging a bad request will be that much harder.

Reply:
Still open?  Reminder, there are 2 things here
- (Smaller) There is no default case.  I'd rather not have the default here in a default not up on l.202.
- (Larger) I'd like to log (trace) the params repo, ref, pth, and which type of operation we picked.